### PR TITLE
docs: Add link to design recipes in root design README

### LINKED ISSUE #754: Reorg recipes for design documentation
**Is your feature request related to a problem? Please describe.**
In our recent engagement, we were searching the playbook for a template to use for design documentation. We tried the following pages and couldn’t find one in here:
1. [Design - Code With Engineering Playbook (microsoft.github.io)](https://microsoft.github.io/code-with-engineering-playbook/design/)
2. Design Documentation section under [Design Reviews - Code With Engineering Playbook (microsoft.github.io)](https://microsoft.github.io/code-with-engineering-playbook/design/design-reviews/)

On digging deeper, we realized the templates for design documentation are under **Recipes** section, but hyperlinked from the [Design Reviews](https://microsoft.github.io/code-with-engineering-playbook/design/design-reviews/) page under the section **Design Review Templates**.

**Describe the solution you'd like**
We think the **Design Review Templates** section would be a better fit under the [Design](https://microsoft.github.io/code-with-engineering-playbook/design/) page as a **Design Documentation Templates** section than under the [Design Review](https://microsoft.github.io/code-with-engineering-playbook/design/design-reviews/) page, since anyone looking for guidance on items to consider while designing and documentation would intuitively look under the Design section.

**Describe alternatives you've considered**
An alternative solution would be to hyperlink the Recipes from **Design Documentation** section in the [Design Reviews](https://microsoft.github.io/code-with-engineering-playbook/design/design-reviews/) page, but that would be mean duplicating the links from the **Design Review Templates** section above.

### PR DESCRIPTION
**Pull request by zmmille2**
# Pull Request Template

## What are you trying to address

Closes #754 
TL;DR - design recipe links are hard to discover because they are nested.

## Description of new changes

@meghapatilcode mentioned that her team initially passed over the design section of the playbook looking for recipes, and asked that recipes be pulled up to the root level.

This change adds a link to the root level of the project for better discoverability.

## For all pull requests

- [X] Link to the issue you are solving (so it gets closed)
- [X] Label the pull request with the appropriate area(s)
- [X] Assign potential reviewers (you may also want to contact them on Teams to ensure timely reviews)
- [X] Assign the project - Engineering Playbook Backlog
- [X] Assign the pull request to yourself

### CODE REVIEW

**Review by charleszipp — APPROVED** [review]

**Review by jakereifer — APPROVED** [review]
The button finally appeared

**Review by jakereifer — APPROVED** [review]

**Review by TessFerrandez — CHANGES_REQUESTED** [review]
See the comment on why a separate link to a sub-section. 
I would like to avoid duplicating information that might become stale, and would like to instead encourage organizing the content such that the sections/content are nicely discoverable when reading from https://microsoft.github.io/code-with-engineering-playbook

File: docs/design/README.md [hunk 1]
```diff
@@ -17,3 +17,7 @@ This covers not only technical design of software, but also architecture design
 - [Design Reviews](design-reviews/README.md)
 - [Non-Functional Requirements Guidance](design-patterns/non-functional-requirements-capture-guide.md)
 - [Sustainable Software Engineering](sustainability/README.md)
+
```
[the remark this answers is not in the document]
Comment (zmmille2) [reply docs/design/README.md:20]: This PR is attempting to address this issue from a playbook consumer.

https://github.com/microsoft/code-with-engineering-playbook/issues/754

It looks (based on the links in the issue) like they are reading the playbook from the github.io site: Do you think making (and linking) the Design Recipes subsection a separate `README` document under `design-reviews` would be a better way to resolve this issue?

[the remark this answers is not in the document]
Comment (zmmille2) [reply docs/design/README.md:20]: @TessFerrandez thank you again for your review, I appreciate it!

I think I've addressed your comment, please let me know if there's anything else you'd like to see changed.

**Review by TessFerrandez — APPROVED** [review]
Looks good to me!

File: docs/design/design-reviews/recipes/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Design Review Recipes
+
+Design reviews come in all shapes and sizes. There are also different items to consider when creating a design at different stages during an engagement
+
+## Design Review Process
+
+- [Incorporate design reviews throughout the lifetime of an engagement](./recipes/engagement-process.md)
```
Comment (TessFerrandez) [inline docs/design/design-reviews/recipes/README.md:7]:
```suggestion
- [Incorporate design reviews throughout the lifetime of an engagement](./engagement-process.md)
```

**Review by TessFerrandez — APPROVED** [review]
